ouroboros-consensus-test/test-consensus/Test/Consensus/MiniProtocol/BlockFetch/Client.hs

```diff
@@ -193,19 +231,52 @@ runBlockFetchTest BlockFetchClientTestSetup{..} = withRegistry \registry -> do
                       case Chain.applyChainUpdates updates chain of
                         Just chain' -> chain'
                         Nothing     -> error "Chain update failed"
-                -- Block until our "ChainSync" thread registered itself to the
-                -- FetchClientRegistry, see 'forkChainSync' below.
-                _ <- blockUntilJust $ Map.lookup peerId <$> readTVar varChains
+                    updateCPS cps =
+                      case CPS.applyChainUpdates (fmap TestHeader <$> updates) cps of
+                        Just cps' -> cps'
+                        Nothing   -> error "Chain update failed"
                 modifyTVar varChains $ Map.adjust updateChain peerId
+                modifyTVar (producerStateVars Map.! peerId) updateCPS
 
         forkChainSync peerId =
-          forkLinkedThread registry ("BracketSync" <> condense peerId) $
+            forkThread registry ("BracketSync" <> condense peerId) $
+            recordChainSyncExceptions $
             bracketSyncWithFetchClient fetchClientRegistry peerId $ do
-              let modifyChains = atomically . modifyTVar varChains
-              bracket_
-                (modifyChains $ Map.insert peerId Chain.Genesis)
-                (modifyChains $ Map.delete peerId)
-                (forkTicking peerId >>= waitThread)
+              let chainSyncTracer =
+                        (\ev -> show peerId <> ": ChainSyncClient: " <> show ev)
+                    >$< tracer
+                  chainDbView = (ChainSyncClient.defaultChainDbView chainDB) {
+                      ChainSyncClient.getCurrentChain =
```

Comment:
Please add a comment explaining why this stub is OK. Why don't we need&want the ChainSync clients to react to the ChainDB's selections? (Moreover does this work only because the `TestBlock` (I'm guessing) has an unlimited forecast range? Ah, and because of the massive `securityParam = SecurityParam 1000` below.)

Reply:
Hmm. I now see the BlockFetch clients' view has the same stub, but there's a comment there. ... Actually, that comment still doesn't clear it up for me. Maybe we can chat about these parts on a call and devise a good explanation for the comments here (CS) and there (BF).

Reply:
I think I did this as the behavior of the clients w.r.t. the changing selected chain seemed orthogonal to this test, but I think we can actually change it such that `getCurrentChain` comes directly from the ChainDB (which I think would resolve your appropriate confusion).

The only thing that needs to be changed/removed is
```haskell
      , counterexample ("Fetched blocks per peer: " <> condense bfcoFetchedBlocks) $
        property $ all (> 0) bfcoFetchedBlocks
```
which no longer holds as e.g. one peer might never have desirable blocks compared to another peer (but all blocks are compared w.r.t. the empty chain).

Reply:
Alternative approaches:

 - tweak the `BlockFetchConsensusInterface` such that all blocks are downloaded.
 - or even add a `testingMode` field for downloading everything in tests